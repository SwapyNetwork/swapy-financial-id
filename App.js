/* @flow */
/* eslint-disable */

import './shim.js'

import React from 'react';
import HomeStack from './app/config/router';
import IdentityProvider from './app/lib/identity';

const App = () => <HomeStack />;

export default App;