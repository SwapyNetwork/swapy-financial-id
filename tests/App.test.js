/* global it, expect, jest */

import * as React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';

jest.mock('react-native-crypto', () => { // eslint-disable-line
  return { randomBytes: size => size * Math.random() };
});

jest.mock('react-native-bip39', () => { // eslint-disable-line
  return { generateMnemonic: async () => Promise.resolve('pÃ£o de batata calzone pizza lanches pedrao pedrinho pedro catupiry peperoni comidas') };
});

jest.mock('../shim', () => { // eslint-disable-line
  return {};
});

it('renders without crashing', () => {
  const rendered = renderer.create(<App />).toJSON();
  expect(rendered).toMatchSnapshot();
});
