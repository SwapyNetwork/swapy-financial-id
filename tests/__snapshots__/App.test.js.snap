// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
    ]
  }
>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "column-reverse",
        },
        Object {
          "backgroundColor": "#000",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        collapsable={undefined}
        pointerEvents="auto"
        style={
          Object {
            "backgroundColor": "#E9E9EF",
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 5,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "flex": 1,
              "width": "100%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#00aeef",
                "fontSize": 48,
                "fontWeight": "bold",
                "top": 40,
              }
            }
          >
            Welcome
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#00aeef",
                "fontSize": 16,
                "marginBottom": 40,
                "marginTop": 25,
              }
            }
          >
            to Swapy Financial ID
          </Text>
          <Image
            source={
              Object {
                "testUri": "../../../app/assets/logo.png",
              }
            }
            style={
              Object {
                "height": 150,
                "marginBottom": 40,
                "width": 150,
              }
            }
          />
          <View
            style={
              Object {
                "paddingLeft": 50,
                "paddingRight": 50,
                "width": "100%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#4a4a4a",
                  "fontSize": 16,
                  "marginBottom": 20,
                  "textAlign": "center",
                }
              }
            >
              Your Financial ID will be linked to a ETH Wallet
            </Text>
            <ActivityIndicator
              animating={true}
              color="#00aeef"
              hidesWhenStopped={true}
              size="large"
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
